### PR TITLE
Removes a lot of lists

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Removes a lot of lists (Like 300k in total). 

## Why It's Good For The Game
Memory optimization and generally faster processing times

## Testing
buildmode plasmafire and crowbarring stuff
## Changelog
NPFC

### CODE REVIEW

File: code/game/turfs/simulated/floor.dm [hunk 1]
```diff
@@ -264,3 +261,12 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
...
 	return !burnt && !broken
+
+/turf/simulated/floor/proc/get_broken_states()
+	return list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5")
+
+/turf/simulated/floor/proc/get_burnt_states()
+	return list("floorscorched1", "floorscorched2")
+
+/turf/simulated/floor/proc/get_prying_tools()
+	return list(TOOL_CROWBAR)
```
Comment (GDNgit) [inline code/game/turfs/simulated/floor.dm:272]:
```suggestion
	return list(TOOL_CROWBAR)

```

File: code/game/turfs/simulated/floor/misc_floor.dm [hunk 2]
```diff
@@ -134,10 +134,14 @@
...
 	floor_tile = /obj/item/stack/tile/noslip
-	broken_states = list("noslip-damaged1","noslip-damaged2","noslip-damaged3")
-	burnt_states = list("noslip-scorched1","noslip-scorched2")
 	slowdown = -0.3
 
+/turf/simulated/floor/noslip/get_broken_states()
+	return list("noslip-damaged1","noslip-damaged2","noslip-damaged3")
+
+/turf/simulated/floor/plating/asteroid/snow/get_burnt_states()
+	return list("noslip-scorched1","noslip-scorched2")
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/misc_floor.dm:143]:
```suggestion
	return list("noslip-damaged1", "noslip-damaged2", "noslip-damaged3")

/turf/simulated/floor/plating/asteroid/snow/get_burnt_states()
	return list("noslip-scorched1", "noslip-scorched2")
```

**Review by DGamerL — APPROVED** [review]

**Review by DGamerL — COMMENTED** [review]
I'm an idiot, that was not supposed to be an approve

File: code/game/turfs/simulated/floor/mineral_floors.dm [hunk 3]
```diff
@@ -95,7 +94,9 @@
 	name = "shuttle floor"
 	icon_state = "titanium"
 	floor_tile = /obj/item/stack/tile/mineral/titanium
-	broken_states = list("titanium_dam1","titanium_dam2","titanium_dam3","titanium_dam4","titanium_dam5")
+
+/turf/simulated/floor/mineral/titanium/get_broken_states()
+	return list("titanium_dam1","titanium_dam2","titanium_dam3","titanium_dam4","titanium_dam5")
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/mineral_floors.dm:99]:
```suggestion
	return list("titanium_dam1", "titanium_dam2", "titanium_dam3", "titanium_dam4", "titanium_dam5")
```

File: code/game/turfs/simulated/floor/mineral_floors.dm [hunk 4]
```diff
@@ -132,7 +133,9 @@
 	name = "shuttle floor"
 	icon_state = "plastitanium"
 	floor_tile = /obj/item/stack/tile/mineral/plastitanium
-	broken_states = list("plastitanium_dam1","plastitanium_dam2","plastitanium_dam3","plastitanium_dam4","plastitanium_dam5")
+
+/turf/simulated/floor/mineral/plastitanium/get_broken_states()
+	return list("plastitanium_dam1","plastitanium_dam2","plastitanium_dam3","plastitanium_dam4","plastitanium_dam5")
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/mineral_floors.dm:138]:
```suggestion
	return list("plastitanium_dam1", "plastitanium_dam2", "plastitanium_dam3", "plastitanium_dam4", "plastitanium_dam5")
```

**Review by DGamerL — APPROVED** [review]
LGTM

**Review by Henri215 — APPROVED** [review]

**Pull request merged**